File: SingletonJava/src/main/java/org/example/LogHelper.java
package org.example;

import java.time.LocalTime;

public class LogHelper {
    private static LogHelper instance;

    private int logStamp= 0;

    private LogHelper() {
        String message = "Stamp\tTime\t\t\tType\tMessage";
        System.out.println(message);
    }

    public static LogHelper getInstance() {
        if (instance == null) {
            instance = new LogHelper();
        }
        return instance;
    }

    public void printLogMessage(LogType logType, String logMessage) {
        logStamp++;
        String logTime = LocalTime.now().toString();
        String message = logStamp + "\t\t" + logTime + "\t" + logType + "\t" + logMessage;
        System.out.println(message);
    }
}

File: SingletonLoggerKotlin/src/main/kotlin/LogHelper.kt
import java.time.LocalTime

object LogHelper {
    private var logStamp = 0
    init {
        val message = "Stamp\tTime\tMessage"
        println(message)
    }
    fun printLogMessage(logMessage: String) {
        logStamp++
        val logTime = LocalTime.now().toString()
        val message = logStamp.toString() + "\t" + logTime + "\t" + logMessage
        println(message)
    }
}